pkg/kongstate/credentials.go

```diff
@@ -130,6 +130,9 @@ func NewOauth2Credential(config interface{}) (*Oauth2Credential, error) {
 	if res.ClientID == nil {
 		return nil, fmt.Errorf("oauth2 is invalid: no client_id")
 	}
+	if res.Name == nil {
```

Comment:
Do we apply or need to apply any name convention ? 

Fo example, k8s resource generically follow this name convention https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

Reply:
No. These are Kong fields inside the K8S object and don't have restrictions beyond being strings.